docs/architecture/accounts.md

```diff
@@ -1,211 +1,123 @@
 
-Functions exposed by the account have the following properties:
+The storage mode is chosen during account creation, it cannot be changed later.
 
-* Functions are actually roots of [Miden program MASTs](https://0xpolygonmiden.github.io/miden-vm/user_docs/assembly/main.html) (i.e., a `32`-byte hash). Thus, the function identifier is a commitment to the code which is executed when a function is invoked.
-* Only account functions have [mutable access](transactions/contexts.md) to an account's storage and vault. Therefore, the only way to modify an account's internal state is through one of the account's functions.
-* Account functions can take parameters and can create new notes.
+## Account core components
 
-> **Note**
-> Since code in Miden is expressed as MAST, every function is a commitment to the underlying code. The code cannot change unnoticed to the user because its hash would change. Behind any MAST root there can only be `256` functions.
+An `Account` is composed of several core components, illustrated below:
 
-#### Example account code
-
-Currently, Miden provides two standard implementations for account code.
-
-##### Basic user account
-
-There is a standard for a basic user account. It exposes three functions via its interface.
-
-<details>
-  <summary>Basic user account code</summary>
-
-  ```arduino
-    use.miden::contracts::wallets::basic->basic_wallet
-    use.miden::contracts::auth::basic
-
-    export.basic_wallet::receive_asset
-    export.basic_wallet::create_note
-    export.basic_wallet::move_asset_to_note
-    export.basic::auth_tx_rpo_falcon512
-  ```
-</details>
-
-[Note scripts](notes.md#the-note-script) or transaction scripts can call `receive_asset`, `create_note`, and `move_asset_to_note` procedures.
-
-Transaction scripts can also call `auth_tx_rpo_falcon512` to authenticate the transaction.
-
-> **Warning**
-> Without correct authentication, i.e., knowing the correct private key, a note cannot successfully invoke `receive_asset`, `create_note`, or `move_asset_to_note`.
-
-##### Basic fungible faucet (faucet for fungible assets)
-
-There is also a standard for a [basic fungible faucet](https://github.com/0xPolygonMiden/miden-base/blob/main/miden-lib/asm/miden/contracts/faucets/basic_fungible.masm).
+<p style="text-align: center;">
+    <img src="../img/architecture/account/account-definition.png" style="width:30%;" alt="Account diagram"/>
+</p>
 
-<details>
-  <summary>Fungible faucet code</summary>
+These components are:
 
-  ```arduino
-  #! Distributes freshly minted fungible assets to the provided recipient.
-  #!
-  #! ...
-  export.distribute
-      # get max supply of this faucet. We assume it is stored at pos 3 of slot 1
-      push.METADATA_SLOT exec.account::get_item drop drop drop
-      # => [max_supply, amount, tag, note_type, RECIPIENT, ...]
+1. [ID](#id)  
+2. [Storage](#storage)  
+3. [Nonce](#nonce)  
+4. [Vault](#vault)  
+5. [Code](#code)
```

Comment:
I would probably change the sequence (both here and in the sections below) to be:

1. ID
2. Code
3. Storage
4. Vault
5. Nonce